Subject: [PATCH] remove revision to support turbine version of clickhouse
---
Index: conn_http.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/conn_http.go b/conn_http.go
--- a/conn_http.go	(revision b24794b0df20877def2a42061fd22c393486ed14)
+++ b/conn_http.go	(date 1777299020021)
@@ -209,9 +209,9 @@
 		client: &http.Client{
 			Transport: rt,
 		},
-		url:             u,
-		revision:        ClientTCPProtocolVersion, // Preflight uses hardcoded revision, may break older versions.
-		encodeRevision:  0,                        // Encoding data over HTTP must use 0. client_protocol_version does not apply to inserts.
+		url: u,
+		// revision:        ClientTCPProtocolVersion, // Preflight uses hardcoded revision, may break older versions.
+		// encodeRevision:  0,                        // Encoding data over HTTP must use 0. client_protocol_version does not apply to inserts.
 		buffer:          new(chproto.Buffer),
 		compression:     compression.Method,
 		blockCompressor: compress.NewWriter(compress.Level(compression.Level), compress.Method(compression.Method)),
@@ -219,12 +219,12 @@
 		blockBufferSize: opt.BlockBufferSize,
 	}

-	handshake, err := conn.queryHello(ctx, func(nativeTransport, error) {})
-	if err != nil {
-		return nil, fmt.Errorf("failed to query server hello: %w", err)
-	}
-	conn.handshake = handshake
-	conn.revision = conn.handshake.Revision
+	//handshake, err := conn.queryHello(ctx, func(nativeTransport, error) {})
+	//if err != nil {
+	//	return nil, fmt.Errorf("failed to query server hello: %w", err)
+	//}
+	//conn.handshake = handshake
+	// conn.revision = conn.handshake.Revision
 
 	return &conn, nil
 }
